setup/src/Magento/Setup/Module/Di/Code/Scanner/PhpScanner.php

```diff
@@ -53,15 +53,28 @@ public function __construct(Log $log, TypeProcessor $typeProcessor = null)
     protected function _findMissingClasses($file, $classReflection, $methodName, $entityType)
     {
         $missingClasses = [];
-        if ($classReflection->hasMethod($methodName)) {
-            $constructor = $classReflection->getMethod($methodName);
-            $parameters = $constructor->getParameters();
-            /** @var $parameter \ReflectionParameter */
-            foreach ($parameters as $parameter) {
-                preg_match('/\[\s\<\w+?>\s([\w\\\\]+)/s', $parameter->__toString(), $matches);
-                if (isset($matches[1]) && substr($matches[1], -strlen($entityType)) == $entityType) {
-                    $missingClassName = $matches[1];
-                    if ($this->shouldGenerateClass($missingClassName, $entityType, $file)) {
+        if (!$classReflection->hasMethod($methodName)) {
+            return $missingClasses;
+        }
+
+        $factorySuffix = '\\' . ucfirst(FactoryGenerator::ENTITY_TYPE);
```

Comment:
If we put factories in this method it becomes less generic.

Reply:
It was not generic before refactoring. This method is used only to get missed factories. I do not want to rename it as it's a protected method.